openelevationservice/server/api/querybuilder.py

```diff
@@ -31,6 +33,91 @@ def _getModel(dataset):
+    """
+    Performs PostGIS query to enrich a polygon geometry.
+    
+    :param geometry: Input 2D polygon to be enriched with elevation
+    :type geometry: Shapely geometry
+    
+    :param format_out: Specifies output format. One of ['geojson', 'polygon']
+    :type format_out: string
+    
+    :param dataset: Elevation dataset to use for querying
+    :type dataset: string
+    
+    :raises InvalidUsage: internal HTTP 500 error with more detailed description. 
+        
+    :returns: 3D polygon as GeoJSON or WKT
+    :rtype: string
+    """
+    
+    Model = _getModel(dataset)
+    
+    if geometry.geom_type == 'Polygon':
+        query_geom = db.session \
+                            .query(func.ST_SetSRID(func.ST_PolygonFromText(geometry.wkt), 4326) \
+                            .label('geom')) \
+                            .subquery().alias('pGeom')
+
+        result_pixels = db.session \
+                            .query(func.ST_PixelAsPoints(
+                                func.ST_Clip(Model.rast, 1, query_geom.c.geom, coord_precision), #.geom
+                                1)) \
+                            .select_from(query_geom.join(Model, ST_Intersects(Model.rast, query_geom.c.geom))) \
+                            .all() #.subquery().alias('getsubrast')
+        
+        point_col, height_col = format_PixelAsPoints(result_pixels)
+        
+        #
+        #query_getelev = db.session \
+        #                    .query(query_subrast.c.geom,
+        #                           query_subrast.c.val) \
+        #                    .subquery().alias('getelev')
+
+        query_points3d = db.session \
+                            .query(func.ST_SetSRID(func.ST_MakePoint(ST_X(point_col),
+                                                                     ST_Y(point_col),
+                                                                     height_col),
+                                              4326).label('geom')) \
+                            .order_by(ST_X(point_col), ST_Y(point_col)) \
+                            .subquery().alias('points3d')
+
+        if format_out == 'geojson':
+            # Return GeoJSON directly in PostGIS
+            query_final = db.session \
+                              .query(func.ST_AsGeoJson(func.ST_Collect(query_points3d.c.geom)))
+            
+        else:
+            # Else return the WKT of the geometry
+            query_final = db.session \
+                              .query(func.ST_AsText(func.ST_MakeLine(query_points3d.c.geom)))
+    else:
+        raise InvalidUsage(400, 4002, "Needs to be a Polygon, not a {}!".format(geometry.geom_type))
+
+    # Behaviour when all vertices are out of bounds
+    if query_final[0][0] == None:
+        raise InvalidUsage(404, 4002,
+                           'The requested geometry is outside the bounds of {}'.format(dataset))
+        
+    return query_final[0][0]
```

Comment:
Is query_final[0][0] the first element of the first result or what it is? Could you assign the result to a variable with a descriptive name, like result_geom or maybe use some function like scalar()? https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.scalar

Reply:
DONE, `query_final.scalar()` is fine.